spec/models/ability_spec.rb

```diff
@@ -0,0 +1,121 @@
+require 'spec_helper'
+require 'cancan/matchers'
+
+describe Ability do
+  let(:subject) { described_class.new(user) }
+  let(:user) { mock_user }
+  let(:item) { Dor::Item.new(pid: 'x') }
+
+  context 'as an administrator' do
+    let(:user) { mock_user(is_admin?: true)}
+
+    it { should be_able_to(:manage, :everything) }
+  end
+
+  context 'as a manager' do
+    let(:user) { mock_user(is_manager?: true)}
+
+    it { should_not be_able_to(:manage, :everything) }
+    it { should be_able_to(:manage_item, item) }
+    it { should be_able_to(:manage_content, item) }
+    it { should be_able_to(:manage_desc_metadata, item) }
+    it { should be_able_to(:create, Dor::AdminPolicyObject) }
+  end
+
+  context 'as a viewer' do
+    let(:user) { mock_user(is_viewer?: true)}
+
+    it { should_not be_able_to(:manage_item, item) }
+    it { should_not be_able_to(:manage_content, item) }
+    it { should_not be_able_to(:manage_desc_metadata, item) }
+    it { should be_able_to(:view_item, item) }
+    it { should be_able_to(:view_metadata, item) }
+    it { should be_able_to(:view_content, item) }
+  end
+
+  context 'for items without an APO' do
+    it { should_not be_able_to(:manage_item, item) }
+    it { should_not be_able_to(:manage_content, item) }
+    it { should_not be_able_to(:manage_desc_metadata, item) }
+    it { should_not be_able_to(:view_item, item) }
+    it { should_not be_able_to(:view_content, item) }
+    it { should_not be_able_to(:view_metadata, item) }
+  end
```

Comment:
Yes, basically I'm looking for equivalent user specs to those in https://github.com/sul-dlss/argo/pull/595 where it considers cases for an APO with or without a governing APO.  So, the thing about issue #76 is that when the object (or 'item') is an APO, it should be able to grant permission to act on itself.  I'm not clear about what the 'item' is exactly in this particular context.